src/rgw/rgw_putobj_processor.cc

```diff
@@ -74,6 +74,18 @@ static int process_completed(const AioResultList& completed, RawObjSet *written)
   return error.value_or(0);
 }
 
+void RadosWriter::add_write_hint(librados::ObjectWriteOperation& op) {
+  const rgw_obj obj = head_obj->get_obj();
+  const RGWObjState *obj_state = obj_ctx.get_state(obj);
```

Comment:
this is called for every rados write, so it's unfortunate that we need to lock this `obj_ctx` every time

ideally the `RadosWriter` constructor would lock it once and save the result in a member variable, but that won't work because `RGWPutObj::execute()` creates this writer before it decides whether to add a compression filter

@dang is planning to merge this `RGWObjState` into `rgw::sal::Object` eventually, which will remove the locking. so it's probably fine to leave this as is for now, just with a `// TODO: remove this lock` comment

what do you guys think?